File: ios/MerchantBaseApp/AppVersionModule.swift
import Foundation
import React

@objc(AppVersionModule)
class AppVersionModule: NSObject {
  
  @objc
  static func requiresMainQueueSetup() -> Bool {
    return false
  }
  
  @objc
  func getVersion(_ resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) {
    if let version = Bundle.main.infoDictionary?["CFBundleShortVersionString"] as? String {
      resolve(version)
    } else {
      reject("VERSION_ERROR", "Failed to get app version", nil)
    }
  }
  
  @objc
  func getBuildNumber(_ resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) {
    if let buildNumber = Bundle.main.infoDictionary?["CFBundleVersion"] as? String {
      resolve(buildNumber)
    } else {
      reject("BUILD_ERROR", "Failed to get build number", nil)
    }
  }
}

@objc
class AppVersionModuleBridge: NSObject {
  @objc
  static func createModule() -> AppVersionModule {
    return AppVersionModule()
  }
}

File: android/app/src/main/java/com/solusinegeri/merchant/app/AppVersionModule.kt
package com.solusinegeri.merchant.app

import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.bridge.ReactContextBaseJavaModule
import com.facebook.react.bridge.ReactMethod
import com.facebook.react.bridge.Promise
import android.content.pm.PackageManager

class AppVersionModule(reactContext: ReactApplicationContext) : ReactContextBaseJavaModule(reactContext) {

    override fun getName(): String {
        return "AppVersionModule"
    }

    @ReactMethod
    fun getVersion(promise: Promise) {
        try {
            val packageManager = reactApplicationContext.packageManager
            val packageName = reactApplicationContext.packageName
            val packageInfo = packageManager.getPackageInfo(packageName, 0)
            val versionName = packageInfo.versionName ?: "1.0.0"
            promise.resolve(versionName)
        } catch (e: Exception) {
            promise.reject("VERSION_ERROR", "Failed to get app version", e)
        }
    }

    @ReactMethod
    fun getBuildNumber(promise: Promise) {
        try {
            val packageManager = reactApplicationContext.packageManager
            val packageName = reactApplicationContext.packageName
            val packageInfo = packageManager.getPackageInfo(packageName, 0)
            val versionCode = packageInfo.longVersionCode
            promise.resolve(versionCode.toString())
        } catch (e: Exception) {
            promise.reject("BUILD_ERROR", "Failed to get build number", e)
        }
    }
}

